presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -160,6 +163,26 @@ public BackgroundHiveSplitLoader(
         this.partitions = new ConcurrentLazyQueue<>(requireNonNull(partitions, "partitions is null"));
         this.hdfsContext = new HdfsContext(session, table.getDatabaseName(), table.getTableName());
         this.schedulerUsesHostAddresses = schedulerUsesHostAddresses;
+        createPathFilter(inputPathFilterClass);
```

Comment:
Can we do in the following way? 
```
this.inputPathFilter = Optional.ofNullable(inputPathFilterClass).map(this::createPathFilter)
```

And thus maybe we could also make `createPathFilter` static.

Reply:
sure.